weave/trace/vals.py

```diff
@@ -294,19 +296,34 @@ def rows(self) -> list[dict]:
                 and self._prefetched_rows is not None
             )
             if should_local_iter:
-                self._rows = list(self._local_iter_with_remote_fallback())
+                self._rows = ThreadSafeInMemorySequence(
```

Comment:
If you substitute "LazyList" in here, I think it becomes pretty clear what we're trying to do

Reply:
yep... `ThreadSafeInMemoryLazyList`